cmd/otelcol/Dockerfile

```diff
@@ -37,6 +37,16 @@ RUN if [ "$ARCH" = "amd64" ]; then \
     fi
 RUN find /usr/lib/splunk-otel-collector/agent-bundle -name "*.key" -delete -or -name "*.pem" -delete
 
+FROM debian:10.11 as journal
+RUN apt update
+RUN apt install -y systemd
+# emptyfile is used as a work-around for copying journald libs for multi-arch
+RUN touch /emptyfile
```

Comment:
Sorry I don't get why this emptyfile workaround is needed. Could you please elaborate?

Reply:
@dmitryax Depending on the architecture, journald file names/ paths can be different, so we need a way to be able to copy the correct files based on its arch. Since If/ Else cant be used as its not supported inside dockerfile, and Docker COPY doesn’t like it when either the file it’s trying to copy is missing, or nothing is provided for it to copy, so added this empty file (a File which we know will exist in both arch) to trick Docker into copying the right set of files. 

Eg: Below ld-linux lib -- if we dont add a file which exist for sure, Docker image would fail complaining about a missing file. 

# Journald libs for ARCH=amd64
COPY --from=journal /emptyfile /lib64/ld-linux-x86-64.so.2* /lib64/

# Journald libs for ARCH=arm64
COPY --from=journal /emptyfile /lib/ld-linux-aarch64.so.1* /lib/

Please let me know if any questions. Thanks!

Reply:
Ok looks good to me, thanks for clarification